mlflow/utils/uri.py

```diff
@@ -256,6 +257,22 @@ def append_to_uri_path(uri, *paths):
     return prefix + urllib.parse.urlunparse(new_parsed_uri)
 
 
+def append_to_uri_query_params(uri, *query_params: Tuple[Any]):
+    """
+    Appends the specified query parameters to an existing URI.
+
+    :param uri: The URI to which to append query parameters.
+    :param query_params: Query parameters to append. Each parameter should
+                         be a 2-element tuple. For example, ``("key", "value")``.
+    """
+    parsed_uri = urllib.parse.urlparse(uri)
+    parsed_query = urllib.parse.parse_qsl(parsed_uri.query)
+    new_parsed_query = parsed_query + list(query_params)
+    new_query = urllib.parse.urlencode(new_parsed_query)
+    new_parsed_uri = parsed_uri._replace(query=new_query)
+    return urllib.parse.urlunparse(new_parsed_uri)
```

Comment:
/ai correct the type hint on `query_params`

Reply:
/ai add the return type hint to this function

Reply:
We can use `Tuple[str, Any]` because the key should be string.

Reply:
Actually, dict is more intuititve for `query_params`. If it's a tuple, it can contain duplicates.

Reply:
Query params can and often do contain multiple values for the same key :)

Reply:
Fixed the type hints :)